polling/src/main/java/io/atleon/polling/reactive/PollingReceiver.java

```diff
@@ -0,0 +1,31 @@
+package io.atleon.polling.reactive;
+
+import reactor.core.publisher.Flux;
+
+/**
+ * The main entry point into the Polling backed stream.
+ * @param <K> The type of the objects in the stream.
+ * @param <V> The type of the object used for acks and nacks.
+ */
+public interface PollingReceiver<K extends PollingRecord<V>, V> {
```

Comment:
Do we actually need both the `PollingReceiver` and the `PollingHandler`? I feel like their responsibilities are overlapping, and there doesn't seem to be a lot of code in the `PollingReceiverImpl` that gives it responsibilities that I can't see belonging to the `PollingHandler` (just looking for possible simplification)

Reply:
The PollingReceiver does a couple things, when there's a failure and re-subscription it cleans up the existing handler and assigns a new instance so we don't have to deal with bad state sticking around. Also it converts the `Flux<Collection<K>>` to `Flux<ReceiverRecord<K>>`. I kind of like having the separation concerns in the logic like this, also the state management is nice.